js/productcat.js

```diff
@@ -169,14 +177,13 @@
    * Sends a GET request.
    */
   const sendGetRequest = (path, params, options, callback, failure) => {
-    const endpoint = params.country == "CN" ? CN_SERVER_URL : SERVER_URL;
+    const endpoint = getEndPoint(params);
     params.cid =settings.cid;
-    params.uid =settings.uid || generateUUID();
+    params.uid =settings.uid;
```

Comment:
the uid seems to be always overriden. We need to support the option where user can provide their own uid, rather than force them to use our auto generated UUID

Reply:
also should check if uid is generated, if not, generate and store into localStorage. This is in case the init fail for whatever reason.

Reply:
changed